agentos/component.py

```diff
@@ -38,26 +38,28 @@ def __init__(
         self,
         repo: Repo,
         identifier: ComponentIdentifier,
-        class_name: str,
         file_path: str,
+        class_name: str = None,
+        instantiate: bool = None,
```

Comment:
I agree it is a bit confusing, so I just changed the default value of `instantiate` to `False`.

A little more background/context on my thinking here:

I though about that and originally hesitated to change the default setting from `True` to `False` since that meant updating all of our example/test registries/specs that omit `instantiate` from their spec (assuming omission implies `True`)... because I was aiming to keep the current default behavior _when `class_name` is provided_.

So then I thought about making the default `True`. However, now that `class_name` is optional, I don't think it makes sense for `instantiate` to default to `True` when `class_name` is `None` (i.e., for a Module Component).

That's what had led me to making `instantiate`'s default value conditional on class_name, which is the implementation you commented on.

After reading your comment though, I decided to just bite the bullet and make the default value `False` which officially makes Components more of a managed Module by default, with the option to add a `class_name` attribute to make the Component a managed Class, and the option to _**additionally**_  set `instantiate` to `True` to make the Component a managed Class Instance.

I've also updated the Component docstring summary of what a Component is and also the param descriptions.

Reply:
I like this choice; I think it makes sense in terms of Components as more general object managers (even though it does require a bit of re-wiring of what we think of as the default).